### DISCUSSION TITLE
CDN image transforms for the backend

### DISCUSSION THREAD

**Discussion by josh-attwood** [craftcms/cms#16197] [Ideas]
On an image heavy website the CMS can use quite a lot of resources and extra space in generating thumbnails for the backend. I pretty much always use an image optimising CDN (Twicpics) with lots of alternatives also out there. It would be great if there was the option to disable Craft's image transforms and add a path structure to generate thumbnail size images through the CDN. This could be a great config option!

**Comment by mmikkel** [comment]
This is already possible via the [`Assets::EVENT_DEFINE_THUMB_URL` event](https://docs.craftcms.com/api/v5/craft-services-assets.html#event-define-thumb-url), which can be used to override how Craft generates its CP thumbnails. 

As an example, [here's how this is implemented in Imager X](https://github.com/spacecatninja/craft-imager-x/blob/6819327eb39c5bfe0be4631f6b894edaa36d65fa/src/ImagerX.php#L400C38-L400C68).

(Unfortunately, AFAIK there is no Twicpics transformer available for Imager, otherwise you could just use that 😃 There are [a few other options](https://plugins.craftcms.com/search?q=imager+transformer&tab=plugins&craft5), though.)
